packages/coinstac-ui/app/render/components/projects/project-covariates-mapper.js

```diff
@@ -0,0 +1,132 @@
+import React, { Component, PropTypes } from 'react';
+import {
+  ControlLabel,
+  FormControl,
+  FormGroup,
+  Table,
+} from 'react-bootstrap';
+import { tail } from 'lodash';
+
+import ComputationFieldCovariates from '../computation-field-covariates';
+
+export default class ProjectCovariatesMapper extends Component {
+  renderHeadingRow() {
+    const {
+      covariates,
+      csv: [row],
+      metaCovariateMapping,
+      metaCovariateErrors,
+      onMapCovariate,
+    } = this.props;
+    const options = [<option disabled key={0} value="null">Map column…</option>]
+      .concat(covariates.map(({ name, type }, i) => {
+        return (
+          <option key={i + 1} value={i}>
+            {`${name} (${ComputationFieldCovariates.typeMap.get(type)})`}
+          </option>
+        );
+      }));
+
+    const tableHeadings = [<th key={0}>{row[0]}</th>].concat(
+      tail(row).map((heading, i) => {
+        const index = i + 1;
```

Comment:
We assume the left column is always file names, but we don’t ignore it in the index count. _Maintain that index!_